pygeo/DVGeometryVSP.py

```diff
@@ -384,7 +452,6 @@ def getVarNames(self):
         """
         return list(self.DVs.keys())
```

Comment:
Good call. Will make that change in a bit

Reply:
I ran into an issue in this process: The `self.DVs` is an ordered dictionary. The order is set by the order the DVs are added in the script. We have been assuming that this is the consistent across processors. May not be the best assumption but it has been true so far. 

This order determines the layout of the Jacobian, and the order of how we perturb the DVs. I cannot just sort the DVs at a point later, because we cannot guarantee the user will not add more DVs. 

So I can return `sorted( list( self.DVs.keys() ) )` here. But I will need to keep the ordering at https://github.com/mdolab/pygeo/blob/ec3b617f537891088af9f528877aa00d103c5923/pygeo/DVGeometryVSP.py#L938 

@joanibal, do you think there will be an issue with the ordering being different in the internal Jacobian vs the return value of `getVarNames`? To me, this is fine since this Jacobian is only used internally, and the sensitivities are communicated in dictionaries. But I may be missing a detail here.

Reply:
I forgot that `self.DVs` was an ordered dict. Because it is ordered line 453 should be fine. 

I think the important thing is to make sure the keys are added to `self.DVs` in a consistent order. 
So maybe we should sort the keys from the user-supplied `dvDict` in `setDesignVars`.

Reply:
Entries are added to self.DVs as the main runscript adds VSP variables to this DVGeo. In almost all scripts, this will be ordered the same on all procs, but I can see someone messing up the order.

However, I think with the general DVGeometry API, we do not really assume the user is ever done with adding additional DVs. i.e. the setDesignVars call does not need to indicate there won't be any more DVs. Although, the setDesignVars call kinda implies the addition of DVs are finished.

Finally, I think other DVGeos also rely on the top level script adding the DVs in a consistent order across processors. So I guess this is okay after all. What do you think?

Reply:
OK. I see what your saying. 

I think it is fine after all